### PR TITLE
Cycamore

### PR DESCRIPTION
**Pull request by scopatz**
Initial recipe for cycamore, which uses cyclus as its main dependency.

### CODE REVIEW

**Comment by scopatz** [comment]
Anyone know why cyclus would be missing symbols from the STL on linux? Is this a problem with how cyclus was compiled or this recipe?  Do I need to add the gcc and libgcc packages?

```
/opt/conda/envs/_build_placehold_placehold_placehold_placehold_placehold_placeho/lib/libcyclus.so: undefined reference to `std::invalid_argument::~invalid_argument()@GLIBCXX_3.4.15'
/opt/conda/envs/_build_placehold_placehold_placehold_placehold_placehold_placeho/lib/libcyclus.so: undefined reference to `std::domain_error::~domain_error()@GLIBCXX_3.4.15'
collect2: error: ld returned 1 exit status
make[2]: *** [bin/cycamore_unit_tests] Error 1
make[2]: Leaving directory `/opt/conda/conda-bld/
```

**Comment by scopatz** [comment]
OK! This build and tests in less than 3 months!

**Comment by scopatz** [comment]
Any chance this could be merged in and / or reviewed?

**Comment by msarahan** [comment]
I don't understand why you duplicated toolchain.  Could you explain your reasoning?  Other than that, I'm fine to merge here.

**Comment by scopatz** [comment]
I was to remove the toolchain by someone (?) for the cyclus, and I simply copied that here.  I could try to implement it here.

**Comment by scopatz** [comment]
Ok, it seems like adding the toolchain back worked!

File: recipes/cycamore/build.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+set -e
+
+if [ "$(uname)" == "Darwin" ]; then
+  # other
+  libext=".dylib"
+  export LDFLAGS="-rpath ${PREFIX}/lib ${LDFLAGS}"
+  export LINKFLAGS="${LDFLAGS}"
+  skiprpath="-DCMAKE_SKIP_RPATH=TRUE"
```
Comment (jakirkham) [inline recipes/cycamore/build.sh]: Oh man. Yeah, I remember learning this the hard way awhile back. Good move.

**Comment by jakirkham** [comment]
Thanks @scopatz.

**Comment by jakirkham** [comment]
> I was to remove the toolchain by someone (?) for the cyclus, and I simply copied that here.  I could try to implement it here.

Would also suggest using the `toolchain` anywhere else that it has been removed.

File: recipes/cycamore/build.sh (hunk 1 above)
Comment (scopatz) [inline recipes/cycamore/build.sh]: Yeah, this was super obscure...

**Comment by scopatz** [comment]
Thanks a million @jakirkham! I remain in your debt :)  And yeah, I have readded the toolchain to cyclus successful

**Pull request merged**